client-app/src/Components/DonatedItemDetails.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { useParams } from 'react-router-dom';
+import PersonIcon from '@mui/icons-material/Person';
+import CategoryIcon from '@mui/icons-material/Category';
+import EventNoteIcon from '@mui/icons-material/EventNote';
+import AssignmentTurnedInIcon from '@mui/icons-material/AssignmentTurnedIn';
+import '../css/DonatedItemDetails.css'; // Import the new CSS file
+
+interface Donor {
+    id: number;
+    firstName: string;
+    lastName: string;
+    contact: string;
+    email: string;
+    addressLine1: string;
+    addressLine2?: string;
+    state: string;
+    city: string;
+    zipcode: string;
+    emailOptIn: boolean;
+}
+
+interface Program {
+    id: number;
+    name: string;
+    description: string;
+    startDate: string;
+    aimAndCause: string;
+}
+
+interface StatusLog {
+    id: number;
+    type: string;
+    dateModified: string;
+}
+
+interface DonatedItem {
+    id: number;
+    itemType: string;
+    currentStatus: string;
+    dateDonated: string;
+    lastUpdated: string;
+    donor: Donor;
+    program: Program;
+    statuses: StatusLog[];
+}
+
+const DonatedItemDetails: React.FC = () => {
+    const { id } = useParams<{ id: string }>();
+    const [donatedItem, setDonatedItem] = useState<DonatedItem | null>(null);
+    const [loading, setLoading] = useState<boolean>(true);
+    const [error, setError] = useState<string>('');
+
+    useEffect(() => {
+        const fetchDonatedItemDetails = async () => {
+            try {
+        
+                const API_BASE_URL = process.env.REACT_APP_BACKEND_API_BASE_URL || '';
+                console.log(API_BASE_URL);
+                console.log(id);
+                const response = await axios.get<DonatedItem>(`${API_BASE_URL}donatedItem/${id}`);
+                
+                console.log(response);
+                setDonatedItem(response.data);
+            } catch (err) {
+                if (axios.isAxiosError(err)) {
+                    setError(`Failed to fetch donated item details: ${err.response?.statusText || 'Server error'}`);
```

Comment:
The error is good at being descriptive, maybe consider making it look a little bit better. Currently it is just a line of kind of small and boring text. Not mandatory, but maybe something to do if you have the time